src/test/testUtil.ts

```diff
@@ -136,21 +136,30 @@ export function installFakeClock(): FakeTimers.InstalledClock {
  * Unlike {@link assertTelemetry}, this function does not do any transformations to
  * handle fields being converted into strings. It will also not return `passive` or `value`.
  */
-export function getMetrics<K extends MetricName>(
-    name: K,
-    ...excludeKeys: string[]
-): readonly Partial<MetricShapes[K]>[] {
-    const query = { metricName: name, excludeKeys: ['awsAccount', 'awsRegion', ...excludeKeys] }
+export function getMetrics<K extends MetricName>(name: K): readonly Partial<MetricShapes[K]>[] {
+    const query = { metricName: name }
 
     return globals.telemetry.logger.query(query) as unknown as Partial<MetricShapes[K]>[]
 }
 
+// Note that Typescript is unable to describe the set of all supertypes of a type.
+// This should be typed something like `asserts actual is * extends T`
+export function partialDeepCompare<T>(actual: unknown, expected: T, message?: string): asserts actual is T {
+    if (typeof actual !== 'object' || !actual || typeof expected !== 'object' || !expected) {
+        assert.deepStrictEqual(actual, expected, message)
+    }
+
+    const partial = selectFrom(actual, ...keys(expected as object))
```

Comment:
does this preserve order, so the assertion still asserts the (potentially non-contiguous) order?

Reply:
Yes it's order-preserving.